backend/enums/activity.js

```diff
@@ -88,6 +89,26 @@ export default class Activity {
     return [...postActivities, ...sharedActivities, ...likedActivities]
   }
 
+  static async retrieveInterests(userId) {
+    const interests = []
+    const unconvertedInterests = await this.getPostsForInterests(userId)
+    await Promise.all(
+      unconvertedInterests.map(async (activity) => {
+        const post = await this.convertToFeedActivity(
+          ActivityType.INTERESTED,
+          activity.id,
+          activity.author,
+          activity.createdAt
+        )
+        interests.push(post)
+      })
+    )
+
+    interests.sort((a, b) => (a.timestamp < b.timestamp ? 1 : -1))
```

Comment:
```suggestion
    const unconvertedInterests = await this.getPostsForInterests(userId)
    const interests = await Promise.all(
      unconvertedInterests.map(async (activity) => (
        const post = await this.convertToFeedActivity(
          ActivityType.INTERESTED,
          activity.id,
          activity.author,
          activity.createdAt
        )
      ))
    )
```
Write it without push so the list is ordered using SQL.